### PR TITLE
DTFS2-4403 - Fix: Portal - Ability to submit a MIA to UKEF (pre approval)

### PR DESCRIPTION
**Pull request by IgnatG**
- The checker should not be able to access the “Abandon” button and they should not be able to change the Bank's reference or the Additional reference.

### CODE REVIEW

File: e2e-tests/gef/cypress/integration/submit-to-ukef-min.spec.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      submitToUkef.comment();
+      submitToUkef.submitButton();
+      submitToUkef.cancelLink();
+    });
+
+    it('submits without comments and displays the confirmation page', () => {
+      submitToUkef.submitButton().click();
+      submitToUkefConfirmation.confirmationPanel().contains('Manual inclusion application submitted to UKEF');
+      submitToUkefConfirmation.dashboardLink();
+    });
```
Comment (ttbarnes) [inline e2e-tests/gef/cypress/integration/submit-to-ukef-min.spec.js:56]: can we add a check to make sure the URL is correct after submission?

File: e2e-tests/gef/cypress/integration/submit-to-ukef.spec.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
-const { format } = require('date-fns');
 import relative from './relativeURL';
 import automaticCover from './pages/automatic-cover';
 import submitToUkef from './pages/submit-to-ukef';
 import submitToUkefConfirmation from './pages/submit-to-ukef-confirmation';
 import applicationDetails from './pages/application-details';
 import CREDENTIALS from '../fixtures/credentials.json';
 
+const { format } = require('date-fns');
```
Comment (ttbarnes) [inline e2e-tests/gef/cypress/integration/submit-to-ukef.spec.js:8]: move to top

File: gef-ui/server/controllers/submit-to-ukef/index.test.js [hunk 3]
```diff
@@ -30,6 +48,10 @@ const MockApplicationResponse = () => {
     createdAt: '1625482095783',
     comment: 'The client needs this asap.',
   }];
+  res.bankId = 'BANKID';
+  res.exporter = mockExporter;
+  res.eligibility = { status: 'Draft' };
+  res.facility = { status: 'Draft' };
```
Comment (ttbarnes) [inline gef-ui/server/controllers/submit-to-ukef/index.test.js:54]: I don't think need eligibility/facility/exporter here now 

if so I think 'Draft' should be 'IN_PROGRESS'

File: gef-ui/templates/partials/submit-to-ukef-confirmation.njk [hunk 4]
```diff
@@ -9,7 +9,9 @@
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds-from-desktop">
       {{ govukPanel({
-        titleText: "Automatic inclusion notice submitted to UKEF",
+        titleText: "Automatic inclusion notice submitted to UKEF"
+        if isAutomaticCover else 
+        "Manual inclusion application submitted to UKEF",
```
Comment (ttbarnes) [inline gef-ui/templates/partials/submit-to-ukef-confirmation.njk:14]: If the `submissionType` text is the same in the DB i'd suggest using that instead: 

```js
titleText: deal.submissionType
```

Then it has less conditions in the template and will work for all types AIN/MIA/MIN

File: gef-ui/server/controllers/submit-to-ukef/index.test.js (hunk 3 above)
Comment (Zainzzkk) [reply gef-ui/server/controllers/submit-to-ukef/index.test.js:54]: fails test if removed

Comment (Zainzzkk) [reply gef-ui/server/controllers/submit-to-ukef/index.test.js:54]: removed

File: gef-ui/templates/partials/submit-to-ukef-confirmation.njk (hunk 4 above)
Comment (Zainzzkk) [reply gef-ui/templates/partials/submit-to-ukef-confirmation.njk:14]: done

**Review by ttbarnes — APPROVED** [review]

File: e2e-tests/gef/cypress/integration/submit-to-checker-min.spec.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import securityDetails from './pages/security-details';
+import applicationDetails from './pages/application-details';
+import applicationSubmission from './pages/application-submission';
+import CREDENTIALS from '../fixtures/credentials.json';
+
+const { format } = require('date-fns');
+
+let applicationId;
+
+context('Submit to UKEF as MIN', () => {
```
Comment (ttbarnes) [inline e2e-tests/gef/cypress/integration/submit-to-checker-min.spec.js:13]: rename e2e test files to MIA

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: e2e-tests/gef/cypress/integration/submit-to-checker-min.spec.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      securityDetails.exporterSecurity().type('test');
+      securityDetails.applicationSecurity().type('test');
+      securityDetails.continueButton().click();
+      securityDetails.visit(applicationId);
+      securityDetails.cancelButton().click();
+
+      applicationDetails.submitButton().click();
+    });
+
+    it('displays correct MIN checker submission message', () => {
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/integration/submit-to-checker-min.spec.js:66]: Rename to MIN to MIA please.

File: e2e-tests/gef/cypress/integration/submit-to-checker-min.spec.js (hunk 5 above)
Comment (Zainzzkk) [reply e2e-tests/gef/cypress/integration/submit-to-checker-min.spec.js:13]: done

File: e2e-tests/gef/cypress/integration/submit-to-checker-min.spec.js (hunk 6 above)
Comment (Zainzzkk) [reply e2e-tests/gef/cypress/integration/submit-to-checker-min.spec.js:66]: done

**Review by ttbarnes — APPROVED** [review]

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**